core/src/main/scala/kafka/log/remote/RemoteLogManager.scala

```diff
@@ -769,23 +789,31 @@ class RemoteLogManager(fetchLog: TopicPartition => Option[Log],
    * @param startingOffset The starting offset to search.
    * @return the timestamp and offset of the first message that meets the requirements. None will be returned if there is no such message.
    */
-  def findOffsetByTimestamp(tp: TopicPartition, timestamp: Long, startingOffset: Long): Option[TimestampAndOffset] = {
```

Comment:
This change was not part of the `2.8x-tiered-storage` branch. Borrowed this code from the PR raised by satishd on trunk

See https://github.com/apache/kafka/pull/11390/files#diff-20bb6d1eb214d04f00a0ce32e758fda01dd5aa0cd300bb6ba11017cd7fe7fa8c

This code will make sure that the offset is looked for only in log segments that are part of the current leader's epoch lineage.